pkg/pillar/cmd/verifier/verifier.go

```diff
@@ -1158,6 +1140,31 @@ func handleModify(ctx *verifierContext, config *types.VerifyImageConfig,
 			status.Safename)
 	}
 
+	if !status.IsContainer && status.State == types.DOWNLOADED {
```

Comment:
What about the case when IsContainer is set?
Should the code be
 if DOWNLOADED {
     if isContainer {
          error
     }
     ...

Driving on when isContainer is set seems like a potential source of (future) problems.

Reply:
For containers, we do not have the SHA - so we don't *yet* do any verification.. Need to revisit this with Avi's change.. Infact, should be part of Avi's change on how they validate when we boot up.

Reply:
That's not my point.

The issue is if you ever get IsContainer && types.DOWNLOADED, what should you do?
I assert you should raise an error there and then, and not treat it as a no-op.

Reply:
I can log an error here.. But not sure what else to do there.

"not treat it as a no-op." --> We should never be in that state. Not sure if there is a point in adding any Error handling code here. The App Instance - if we ever hit this path because of a bug - will get stuck in the error state -- Or in the init state..

I will change this to add an error..

Reply:
Good Catch.. Actually there is a case where Container image is still in Downloader state. And we need to mark it as DELIVERED. Fixed it.. Good catch..

Reply:
Yeah Avi (and I) just ran into that bug in his code today! I don't understand how it worked in current master.